.github/workflows/ci.yml

```diff
@@ -47,10 +34,58 @@ jobs:
 
       - name: Install OPAM dependencies
         run: |
-          opam update
-          opam install -y dune
           opam uninstall -y brr
-          opam install -y . --deps-only --with-test
+          opam install -y . --deps-only --with-test || (opam update && opam install -y . --deps-only --with-test)
+
+      - name: Perform dune build
+        run: opam exec -- dune build
+
+      - name: Perform dune runtest
+        run: opam exec -- dune runtest
+
+  matrix:
```

Comment:
Using the name `matrix` would be confusing as it would be covered by the `matrix` keyword in GitHub Actions.

Reply:
indeed. i'll change it to `matrix-check`.

Reply:
↑ done